xdsl/traits.py

```diff
@@ -222,7 +222,9 @@ def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
         # import builtin here to avoid circular import
         from xdsl.dialects.builtin import StringAttr
 
-        if "sym_name" not in op.attributes:
+        if "sym_name" not in op.attributes or not isinstance(
+            op.attributes["sym_name"], StringAttr
+        ):
             if self.is_optional_symbol(op):
```

Comment:
I think that if you have an optional symbol, and you have `sym_name` of the wrong type, then this will return `None` (and I think we should return an error there).

Reply:
Pushed something else, WDYT?